### ISSUE TITLE
Actual state is written once again when using cover.close_cover

### ISSUE THREAD

**Issue by bcutter** [gluap/pyduofern-hacs#24]
Using v0.4.0 I discovered something new, very confusing:

Closing all covers with an automation sending "cover.close_cover" to a group with all covers.

Based on todays observation, the only one closing fine (open -> automation -> closed) is a 1800:

![image](https://user-images.githubusercontent.com/13799156/214095449-59671492-d3f2-438e-a825-fa7d1983f044.jpeg)

All others (1200 if that makes a pattern) behave different which creates confusion on several places (open -> closing -> **open** -> closed), few examples:

![image](https://user-images.githubusercontent.com/13799156/214095937-85b3cc00-dc8d-4aae-93ce-2df5ca0e1b20.jpeg)

![image](https://user-images.githubusercontent.com/13799156/214095973-428774a3-fc8e-4e6a-b951-4fa7b688c272.jpeg)

![image](https://user-images.githubusercontent.com/13799156/214096005-7a282a5e-0947-48e7-8058-d99d5464db7d.jpeg)

![image](https://user-images.githubusercontent.com/13799156/214096040-2428b9c0-34d0-46d1-a5cf-6503fe92d6e8.jpeg)

**Comment by gluap** [comment]
I have been thinking about this - Do you know whether the shutter was already 100% down while homeassistant showed "wird geöffnet"? I believe the meassage may mean "received an opening state that was not 0%" so it could just be an intermediate state -

My take on this would be:
-> Automation sends close
-> Cover starts closing
-> Cover not yet closed but sends a status update "I'm at 30% closed" (homeassistant interprets this as "my new state is open", because "closed" is state == 0% open and nothing else.
-> some seconds later: Cover closed and sends state 0% open -> Homeassistant interprets this as "my new state is closed".

**Comment by bcutter** [comment]
Not it wasn't. Your assumption is right: the log entry signals the initiated action (closing/opening). It is shown instantly once the action is started.

It's just that this message (and state because this message also triggers the history graph etc.) is totally wrong:

![image](https://user-images.githubusercontent.com/13799156/215285267-a8c9a0d9-acf3-493f-84dc-0057e4b20b1f.jpeg)

Not sure if it's a HA thing or was introduced with 0.4.0 of this integration. In the end it creates wrong states and leads to confusion -.-

**Comment by gluap** [comment]
It might be both - in 0.4.0 i experimentally introduced the "opening" and "closing" states - Duofern had them but I hadn't exposed them to homeassistant  before 0.4.0. Therefor until 0.4.0 the blinds could only be either "open" or "closed" but not "opening" or "closing". But I also observe that these seem to be unreliable on my end. I only have RolloTron 2510 and they get stuck in "opening" or "closing" state very often. Which is bad because that makes homeassistant disable the "open" and "close" buttons until it receives the next state from the blind.

I think I'll have to disable the "opening" and "closing" states again.

**Comment by bcutter** [comment]
> I think I'll have to disable the "opening" and "closing" states again.

To bad as it basically is a nice feature - if it works as intended and does not create that kind of confusion -.-
I would really like to keep that but I'm not sure if that's possible.

> I also think that the "update device state" action (which - as you guessed - tries to update all shutters at once) will have to be changed to only address one blind. My impression is that it often does not work and I suspect that the problem is that the responses come in while the loop is still sending the requests for the reminding blinds and that as a result the blinds all talk over each other and no state is successfully received.

That would be perfect anyway: I still suffer from #18, even it is quite rare. That way I could reenable my manual state fix (as a backup solution) and Fire the update device state" service against single devices, not all of them. That's exactly how I used the "cover.stop_cover" before. See https://github.com/gluap/pyduofern-hacs/issues/18#issuecomment-1371494350 for my thoughts/question on that service. From my side a warm "go" on limiting this service.

**Comment by bcutter** [comment]
I will test v0.4.1 asap.

What about
> Add a low-key auto update by letting the devices schedule an update themselves every five minutes when duofern asks for their state

Does this mean there will be a **constant** radio transmission (from every device, every five minutes, **forever**)?
Or only "once when using the ask-for-update-service on a random time within 5 minutes" (to avoid sending states from multiple devices at the same time)?

I fear having constant radio transmissions in order to workaround the state not reported accurate/at all issue -.- So I'd be happy to better understand how exactly this change is intended to work.

**Comment by gluap** [comment]
Yes, constant means: it will send a short "what's your state" for every device every 5 minutes.

That's about what for instance weather stations (including duofern), thermometers and humidity sensors do in the same 433 MHz frequency band and lawfully it's ok in Europe, see [for instance for Germany](https://www.bundesnetzagentur.de/SharedDocs/Downloads/DE/Sachgebiete/Telekommunikation/Unternehmen_Institutionen/Frequenzen/Allgemeinzuteilungen/FunkanlagenGeringerReichweite/2018_05_SRD_pdf.pdf?__blob=publicationFile&v=7)

I can also disable that again.

**Comment by bcutter** [comment]
> I can also disable that again.

50/50. While it sounds in theory it should solve a few missing states problems, on the other hand it means a constant RF noise (in my case I have many DuoFERN devices).

How about a config option for this? Maybe using two services
- "duofern.enable_constant_state_check"
- "duofern.enable_constant_state_check"
?

**Comment by gluap** [comment]
If  you check the logs after such an occurence my bet is you will find that when "was opened" is displayed the logs show that a message was received by the shutter. The message said something like "my opening state is 50%" (intermediate state between "100% open" and "closed"). Homeassistant sasys "hey 50% is not closed" so the shutter is obviously open now -> "was opened". once it receives the "shutter is 100% closed" message it displays "shutter was closed".

**Comment by bcutter** [comment]
I didn't check it (yet), but you are probably/very likely right. The question is: how can we adjust that behavior. Cause it's quite irritating that an opened shutter is opened again :-D

**Comment by bcutter** [comment]
> I didn't check it (yet), but you are probably/very likely right. The question is: how can we adjust that behavior. Cause it's quite irritating that an opened shutter is opened again :-D

Ah OK that's an answer, in a way :-)

![grafik](https://github.com/gluap/pyduofern-hacs/assets/13799156/b0f1dcbe-40e6-4531-9aba-d422d3ce74a6)
